### PR TITLE
Set HUGINN_AGENT_SPEC_RUNNER

### PR DESCRIPTION
**Pull request by mandrews**
Changes:
- Generate `spec_helper.rb` that allows `rake spec` to run with gemspec development dependencies
- Set `HUGINN_AGENT_SPEC_RUNNER` so that [`huginn/Gemfile`](https://github.com/cantino/huginn/pull/1941/files#diff-8b7db4d5cc4b8f6dc8feb7030baa2478) will load gemspec dependencies when run with `rake spec`

Fixes:
https://github.com/cantino/huginn_agent/issues/13

Depends:
https://github.com/cantino/huginn/pull/1941

### CODE REVIEW

File: lib/huginn_agent/templates/newagent/spec/spec_helper.rb.tt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+base_path = File.expand_path('../../', __FILE__)
```
Comment (dsander) [inline lib/huginn_agent/templates/newagent/spec/spec_helper.rb.tt]: I don't fully understand why this is needed, doesn't running `bundle exec rspec  --pattern ../*_spec.rb` work without it?

Comment (mandrews) [inline lib/huginn_agent/templates/newagent/spec/spec_helper.rb.tt]: @dsander determining the absolute path of the huginn_agent is necessary to run the spec outside of `spec_runner.rb`, e.g.
```bash
$ pwd
my_agent
$ (cd spec/huginn; bundle exec rspec --require ../spec_helper.rb ../*_spec.rb)
```
while `rake spec` is a convenient helper, when you try to set a breakpoint with `byebug`, the `popen3`/[`shell_out`](https://github.com/cantino/huginn_agent/blob/f4edd3f001ab85d6519ba7800bd66f3d878b5698/lib/huginn_agent/spec_runner.rb#L46) caused some STDIN inference.

Comment (mandrews) [inline lib/huginn_agent/templates/newagent/spec/spec_helper.rb.tt]: @dsander you're right this is not necessary - I will remove. This is only necessary in the huginn_agent gem I am currently working on, where I'd like to include [`my_agent/spec/support/**/*`](http://stackoverflow.com/a/20615194/809290):
```ruby
Dir[Rails.root.join("spec/support/**/*.rb")].each {|f| require f}
```

but `Rails.root` is `spec/huginn', so I've included it as:

```ruby
Dir["#{base_path}/spec/support/**/*.rb"].each { |f| require f }
```

**Comment by mandrews** [comment]
@dsander we could move the [`huginn/Gemfile` patch](https://github.com/cantino/huginn_agent/pull/15) to the [`spec_runner.rb`](https://github.com/cantino/huginn_agent/blob/master/lib/huginn_agent/spec_runner.rb) and pass the development dependencies as `ADDITIONAL_GEMS`.  This would work when using `rake spec` to trigger [`SpecRunner#spec`](https://github.com/cantino/huginn_agent/blob/master/lib/huginn_agent/spec_runner.rb#L40). The reason I am suggesting a patch to `huginn/Gemfile` is to enable the use of `rspec` directly outside of `spec_runner.rb`, e.g.:
```bash
(cd spec/huginn; bundle exec rspec --require ../spec_helper.rb ../*_spec.rb)
```

I'd like to be able to use `rspec` directly so that I can set breakpoints via `byebug` and I believe that the `popen3`/[`shell_out`](https://github.com/cantino/huginn_agent/blob/f4edd3f001ab85d6519ba7800bd66f3d878b5698/lib/huginn_agent/spec_runner.rb#L46) causes some STDIN inference ([see also](https://github.com/cantino/huginn_agent/issues/13#issuecomment-287509775)).

File: lib/huginn_agent/templates/newagent/spec/spec_helper.rb.tt (hunk 1 above)
Comment (dsander) [inline lib/huginn_agent/templates/newagent/spec/spec_helper.rb.tt]: I see, so having `spec_helper.rb` inside the gem is needed if you want to use files in `spec/support` and run the specs with `rspec` directly?

**Comment by dsander** [comment]
I think running `rspec` directly would still work when we patch the Huginn `Gemfile` during the setup that is done in the [prepare](https://github.com/cantino/huginn_agent/blob/master/lib/tasks/spec.rake#L11-L14) rake task.

File: lib/huginn_agent/templates/newagent/spec/spec_helper.rb.tt (hunk 1 above)
Comment (mandrews) [inline lib/huginn_agent/templates/newagent/spec/spec_helper.rb.tt]: @dsander yes- exactly

**Comment by mandrews** [comment]
@dsander I'm not seeing an existing patch to `spec/huginn/Gemfile`. I could see the `huginn/Gemfile` patch being applied at `prepare` time if you want to go that route - are you thinking after [SpecRunner#reset](https://github.com/cantino/huginn_agent/blob/master/lib/huginn_agent/spec_runner.rb#L20)?

**Comment by dsander** [comment]
@mandrews Right, either at the end of `reset` or in a new method like `runner.patch`. Since we always hard reset the repository we should not have any problem with conflicts.

File: lib/huginn_agent/templates/newagent/spec/spec_helper.rb.tt (hunk 1 above)
Comment (dsander) [inline lib/huginn_agent/templates/newagent/spec/spec_helper.rb.tt]: Got it, I will have a look if we can keep `STDIN` working with our `open3` approach if we can't I am oke with adding it.

**Comment by mandrews** [comment]
@dsander I encapsulated the `huginn/Gemfile` patch within `runner.patch` and tested with a fresh `bundle exec huginn_agent new my_agent`.

**Comment by dsander** [comment]
Nice, the `Gemfile` changes look good to me.

Would you mind checking out #16? This should allow you to use `pry` when running `rake spec` and remove the need for having a `spec_helper` inside the Agent gem.

**Comment by mandrews** [comment]
Closing in favor of:
https://github.com/cantino/huginn_agent/pull/17

**Pull request closed without merging**